azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -0,0 +1,378 @@
+		sqlPoolInfo.SQLPoolResourceProperties.SourceDatabaseID = utils.String(sourceDatabaseId)
+	}
+
+	future, err := sqlClient.Create(ctx, workspaceId.ResourceGroup, workspaceId.Name, name, sqlPoolInfo)
+	if err != nil {
+		return fmt.Errorf("creating Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, sqlClient.Client); err != nil {
+		return fmt.Errorf("waiting on creating future for Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+	}
+
+	if d.Get("data_encrypted").(bool) {
+		parameter := synapse.TransparentDataEncryption{
+			TransparentDataEncryptionProperties: &synapse.TransparentDataEncryptionProperties{
+				Status: synapse.TransparentDataEncryptionStatusEnabled,
+			},
+		}
+		if _, err := sqlPTDEClient.CreateOrUpdate(ctx, workspaceId.ResourceGroup, workspaceId.Name, name, parameter); err != nil {
+			return fmt.Errorf("setting `data_encrypted`: %+v", err)
+		}
+	}
+
+	resp, err := sqlClient.Get(ctx, workspaceId.ResourceGroup, workspaceId.Name, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+	}
+
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Synapse SqlPool %q (Resource Group %q / workspaceName %q) ID", name, workspaceId.ResourceGroup, workspaceId.Name)
+	}
+
+	d.SetId(*resp.ID)
+	return resourceArmSynapseSqlPoolRead(d, meta)
+}
+
+func resourceArmSynapseSqlPoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	sqlClient := meta.(*clients.Client).Synapse.SqlPoolClient
+	sqlPTDEClient := meta.(*clients.Client).Synapse.SqlPoolTransparentDataEncryptionClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SynapseSqlPoolID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("data_encrypted") {
+		status := synapse.TransparentDataEncryptionStatusDisabled
+		if d.Get("data_encrypted").(bool) {
+			status = synapse.TransparentDataEncryptionStatusEnabled
+		}
+
+		parameter := synapse.TransparentDataEncryption{
+			TransparentDataEncryptionProperties: &synapse.TransparentDataEncryptionProperties{
+				Status: status,
+			},
+		}
+		if _, err := sqlPTDEClient.CreateOrUpdate(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name, parameter); err != nil {
+			return fmt.Errorf("updating `data_encrypted`: %+v", err)
+		}
+	}
+
+	if d.HasChanges("sku_name", "tags") {
+		sqlPoolInfo := synapse.SQLPoolPatchInfo{
+			Sku: &synapse.Sku{
+				Name: utils.String(d.Get("sku_name").(string)),
+			},
+			Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+		}
+
+		if _, err := sqlClient.Update(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name, sqlPoolInfo); err != nil {
```

Comment:
this looks like a Swagger/API bug since this should be a Long Running Operation?

Reply:
after testing the api or trying in the portal, I found only changing sku is a long running operation (in the portal, it's called `scale`). Updating other fields is fast. But there are the same rest api. there is not a standalone api for changing sku.